### PR TITLE
Feat/draft gate

### PR DESCRIPTION
**Pull request by fbenfraj**
# Feat/draft gate

## Type of PR
- [✅ ] - feature

## Included in this PR:
- Gate State Chip component added to GateCards and CategoriesLists
- "More" Popover component for Gate options (delete, edit...)
- Confirmation Dialog component on publish/unpublish or delete
- Create Gate: Loads draft data to the form
- Redirections to correct pages depending on gate_state
- Rename create-navbar to publish-navbar
- Save as Draft logic with validation
- Refactor of the createGate function (fused with saveDraft into handleMutation)
- Added DraftGateTypes type schema (since tasks format is different when we send it to the database `tasks: { data: [...]}` and when we retrieve it `tasks: [ ]`)
- Fixes for long descriptions breaking the visuals and usage of ReadMore
- Filtering unpublished gates on Explore page (TODO: move this logic to backend)
- View Gate: Edit, Publish/Unpublish and Delete logic
- View Gate: Remove instant completion on gate view when there is no tasks on Draft
- Update queries on_conflict and mutations

### CODE REVIEW

**Review by rafagomes — CHANGES_REQUESTED** [review]

**Review by kbooz — CHANGES_REQUESTED** [review]

File: apps/website/components/atoms/more-popover.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+              {!option.hidden && (
+                <Typography
+                  key={`${option.text}-${index}`}
+                  sx={{
+                    p: 2,
+                    cursor: 'pointer',
+                    minWidth: '200px',
+                    '&:hover': { backgroundColor: 'rgba(0, 0, 0, 0.08)' },
+                  }}
+                  onClick={option.action}
```
Comment (kbooz) [inline apps/website/components/atoms/more-popover.tsx:65]: Onclick on a non-button component

File: apps/website/components/atoms/more-popover.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import { Avatar, IconButton, Popover, Typography } from '@mui/material';
+
+const MorePopover = ({ options }) => {
```
Comment (kbooz) [inline apps/website/components/atoms/more-popover.tsx:6]: Missing prop types

File: apps/website/components/molecules/categories-input.tsx [hunk 3]
```diff
@@ -3,12 +3,13 @@ import { Chip, TextField, Autocomplete } from '@mui/material';
 
 import { CATEGORIES } from '../../constants/gate';
 
-export const CategoriesInput = ({ set, ...props }) => {
+export const CategoriesInput = ({ set, defaultValue, ...props }) => {
```
Comment (kbooz) [inline apps/website/components/molecules/categories-input.tsx:6]: Missing prop types

File: apps/website/components/molecules/skills-input.tsx [hunk 4]
```diff
@@ -3,12 +3,13 @@ import { Chip, TextField, Autocomplete } from '@mui/material';
 
 import { SKILLS } from '../../constants/skills';
 
-export const SkillsInput = ({ set, ...props }) => {
+export const SkillsInput = ({ set, defaultValue, ...props }) => {
```
Comment (kbooz) [inline apps/website/components/molecules/skills-input.tsx:6]: Missing prop types

File: apps/website/components/organisms/confirm-dialog/confirm-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+} from '@mui/material';
+
+const ConfirmDialog = (props) => {
```
Comment (kbooz) [inline apps/website/components/organisms/confirm-dialog/confirm-dialog.tsx:9]: missing prop types

File: apps/website/components/organisms/gates/create/gate-published.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+const style: SxProps = {
+  bgcolor: 'background.paper',
+  p: 3,
+  minHeight: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+};
+
+export default function GatePublishedModal({ gateId, open, handleClose }) {
```
Comment (kbooz) [inline apps/website/components/organisms/gates/create/gate-published.tsx:21]: Missing prop types

File: apps/website/components/organisms/tasks-area/tasks-area.tsx [hunk 7]
```diff
@@ -49,39 +51,47 @@ const defaultTaskData = (
   }
 };
 
-const TaskArea = () => {
-  const { control, trigger } = useFormContext<CreateGateTypes>();
+const TaskArea = ({ draftTasks, onDelete }) => {
```
Comment (kbooz) [inline apps/website/components/organisms/tasks-area/tasks-area.tsx:54]: Missing prop types

File: apps/website/pages/gate/new.tsx [hunk 8]
```diff
@@ -1,28 +1,55 @@
+import { redirect } from 'next/dist/server/api-utils';
 import { useRouter } from 'next/router';
 import { useEffect } from 'react';
 
 import { CreateGateTemplate } from '../../components/templates/create-gate';
 import { ROUTES } from '../../constants/routes';
+import { gqlAnonMethods } from '../../services/api';
 
-export default function CreateGate() {
+export default function CreateGate({ gateProps }) {
```
Comment (kbooz) [inline apps/website/pages/gate/new.tsx:9]: Missing prop types

**Review by kbooz — CHANGES_REQUESTED** [review]

File: apps/website/pages/gate/new.tsx (hunk 8 above)
Comment (fbenfraj) [reply apps/website/pages/gate/new.tsx]: Added

File: apps/website/components/organisms/gates/create/gate-published.tsx (hunk 6 above)
Comment (fbenfraj) [reply apps/website/components/organisms/gates/create/gate-published.tsx]: Added

File: apps/website/components/organisms/confirm-dialog/confirm-dialog.tsx (hunk 5 above)
Comment (fbenfraj) [reply apps/website/components/organisms/confirm-dialog/confirm-dialog.tsx]: Added

File: apps/website/components/molecules/skills-input.tsx (hunk 4 above)
Comment (fbenfraj) [reply apps/website/components/molecules/skills-input.tsx]: Added

File: apps/website/components/molecules/categories-input.tsx (hunk 3 above)
Comment (fbenfraj) [reply apps/website/components/molecules/categories-input.tsx]: Added

File: apps/website/components/atoms/more-popover.tsx (hunk 2 above)
Comment (fbenfraj) [reply apps/website/components/atoms/more-popover.tsx]: Added

File: apps/website/components/atoms/more-popover.tsx (hunk 1 above)
Comment (fbenfraj) [reply apps/website/components/atoms/more-popover.tsx:74]: Fixed

**Review by NunoMartins21 — APPROVED** [review]
Seems like everything is good to go

**Pull request merged**